failure_test.go

```diff
@@ -30,7 +32,9 @@ func TestHttp429(t *testing.T) {
 	_, err := ch.c.Get(ctx, testCid)
 	require.Error(t, err)
 
-	ferr := err.(*caboose.ErrSaturnTooManyRequests)
+	var ferr *gateway.ErrorRetryAfter
```

Comment:
This is going to create a circular dependency.

the gateway uses caboose as it's blockstore / fetcher
us going back to the gateway for errors seems like the wrong direction

can we just define a common interface, e.g. if the error exposes a `RetryAfter()` method that we can conform to?

Reply:
But caboose already requires `go-libipfs`: https://github.com/filecoin-saturn/caboose/blob/main/go.mod#L9

Reply:
maybe put another way:
caboose is a common saturn client
why would it be returning gateway-specific errors?

Reply:
I don't feel strongly. If we want to keep caboose consumer-agnostic, it is fine to close this PR, and keep explicit error translation in: https://github.com/ipfs/bifrost-gateway/pull/56/files/d65231f2119edb64a61d6044bd758181a068bb35#diff-4d0eb65b9891ffe23bd03c7596f2b573a5ecd5378e939ab7801acb5dca97b147R87

Reply:
It's also fine by me. We just have to remove the comments in the current caboose code that say the code will be refactored with the gateway errors to avoid future confusion.

Reply:
@hacdias mind refactoring this PR to only remove these comments? :pray:

Reply:
🙏

i suspect as this becomes 'the way to access data from saturn with retries that work' we'll find that all the other clients also realize they also want the logic here as well :)